src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,92 @@
+import java.util.*;
 public class StringManipulation implements StringManipulationInterface {
+    private String pString;
 
     @Override
     public String getString() {
-        return null;
+        return pString;
     }
 
     @Override
     public void setString(String string) {
+        this.pString = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if(pString == null || pString.isEmpty()){
+            return 0;
+        }
+
+        String[] words = pString.split("\\s+"); //split string to words using space
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if(n <= 0){
+            throw new IllegalArgumentException("Input n cannot be 0 or negative");
+        }
+
+        String theString = getString();
+        StringBuilder sb = new StringBuilder(theString);
+
+        if(maintainSpacing){ //replace the nth letter with a space
+            for(int i = n-1; i < theString.length(); i += n){
+                sb.setCharAt(i, ' ');
+            }
+            return sb.toString();
+        }
+        else{
+            if(n == 1){ //return empty string
+                return "";
+            }
+            else{ //delete nth letter without spacing
+                for(int i = n-1; i < theString.length(); i += n){
+                    while (i < sb.length()){
+                        sb.deleteCharAt(i);
+                        i += n - 1;
+                    }
+                }
+                return sb.toString();
+            }
+        }
     }
 
     @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+    public String[] getSubStrings(int startWord, int endWord){
+        if(startWord <= 0 || endWord <= 0 || startWord > endWord){
+            throw new IllegalArgumentException("Invalid startWord or endWord");
+        }
+
+        String[] words = getString().split("\\s+"); //split string to words using space
+
+        if(words.length < endWord){
+            throw new IndexOutOfBoundsException("String has less than 'endWord' words");
+        }
+        //returns a copy of array that only include startWord to endWord
+        return Arrays.copyOfRange(words, startWord - 1, endWord);
     }
 
     @Override
-    public String restoreString(int[] indices) {
-        return null;
-    }
+    public String restoreString(int[] indices){
+        if(indices == null || this.pString == null) {
```

Comment:
May also want to check for string and indices[] with length zero as a precaution